Tasks/Common/webdeployment-common-v2/packageUtility.ts

```diff
@@ -29,12 +29,36 @@ export class Package {
     constructor(packagePath: string) {
         this._path = PackageUtility.getPackagePath(packagePath);
         this._isMSBuildPackage = undefined;
+        let workingDirectory = tl.getVariable("system.defaultworkingdirectory");
+        try {
+            if (workingDirectory && this._path.indexOf(workingDirectory) == 0) {
+                let relativePackagePath = packagePath.substring(workingDirectory.length);
+                if (relativePackagePath.indexOf(path.sep) == 0) {
+                    relativePackagePath = relativePackagePath.substring(1);
+                }
+                let endIndex = relativePackagePath.indexOf(path.sep);
+                endIndex = endIndex >= 0 ? endIndex : relativePackagePath.length;
+                this._artifactAlias = relativePackagePath.substring(0, endIndex);
+            }
+            else {
+                this._artifactAlias = null;
+            }
+            tl.debug("Artifact alias of package is: "+this._artifactAlias);
+        }
+        catch(error) {
+            tl.debug("Error in determining artifact alias of package.");
```

Comment:
Should you print the error message as well?

Reply:
yes will do.